frame/contracts/src/schedule.rs

```diff
@@ -72,7 +73,8 @@ pub const INSTR_BENCHMARK_BATCH_SIZE: u32 = 1_000;
 /// changes are made to its values.
 #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "std", serde(bound(serialize = "", deserialize = "")))]
-#[derive(Clone, Encode, Decode, PartialEq, Eq, ScheduleDebug, DefaultNoBound)]
+#[derive(Clone, Encode, Decode, PartialEq, Eq, ScheduleDebug, DefaultNoBound, TypeInfo)]
+#[scale_info(skip_type_params(T))]
```

Comment:
What happens when we dont skip this? It will include `Config` in the type definition?

Reply:
It means the concrete type of `T` will be `null` in the params list:

```
"params": [
  {
    "name": "T",
    "type": null
  }
],
```

This means we don't require `T` to be constrained to `TypeInfo`, which is not required for e.g. a runtime impllementing`Config` in this case because the type itself is never encoded and just ends up in a `Phantom<T>`.

See more background https://github.com/paritytech/scale-info/pull/96

Reply:
Oh, looks like Basti also pointed this out in his review, we just keep circling back to the same topic heh.

Reply:
This attribute has different to the `bounds` attribute, it seems to be a common misunderstanding, see https://github.com/paritytech/scale-info/pull/96.

I'm not sure what a good solution to the misunderstanding is, perhaps a rename of the attribute.